pl_bolts/models/detection/retinanet/retainanet_module.py

```diff
@@ -0,0 +1,148 @@
+from argparse import ArgumentParser
+from typing import Any, Optional
+
+import pytorch_lightning as pl
+import torch
+
+from pl_bolts.metrics.object_detection import _evaluate_iou
+from pl_bolts.models.detection.retinanet import create_retinanet_backbone
+from pl_bolts.utils import _TORCHVISION_AVAILABLE
+from pl_bolts.utils.warnings import warn_missing_pkg
+
+if _TORCHVISION_AVAILABLE:
+    from torchvision.models.detection.retinanet import RetinaNet as torchvision_RetinaNet
+    from torchvision.models.detection.retinanet import retinanet_resnet50_fpn, RetinaNetHead
+else:  # pragma: no cover
+    warn_missing_pkg("torchvision")
+
+
+class RetinaNet(pl.LightningModule):
+    """
+    PyTorch Lightning implementation of Retina Net `Focal Loss for
+    Dense Object Detection <https://arxiv.org/abs/1708.02002>`_.
+
+    Paper authors: Tsung-Yi Lin, Priya Goyal, Ross Girshick, Kaiming He, Piotr Dollár
+
+    Model implemented by:
+        - `Aditya Oke <https://github.com/oke-aditya>`
+
+    During training, the model expects both the input tensors, as well as targets (list of dictionary), containing:
+        - boxes (`FloatTensor[N, 4]`): the ground truth boxes in `[x1, y1, x2, y2]` format.
+        - labels (`Int64Tensor[N]`): the class label for each ground truh box
+
+    CLI command::
+
+        # PascalVOC
+        python retinanet_module.py --gpus 1 --pretrained True
+    """
+
+    def __init__(
+        self,
+        learning_rate: float = 0.0001,
+        num_classes: int = 91,
+        backbone: Optional[str] = None,
+        fpn: bool = True,
+        pretrained: bool = False,
+        pretrained_backbone: bool = True,
+        trainable_backbone_layers: int = 3,
+        **kwargs: Any,
+    ):
+        """
+        Args:
+            learning_rate: the learning rate
+            num_classes: number of detection classes (including background)
+            backbone: Pretained backbone CNN architecture.
+            fpn: If True, creates a Feature Pyramind Network on top of Resnet based CNNs.
+            pretrained: if true, returns a model pre-trained on COCO train2017
+            pretrained_backbone: if true, returns a model with backbone pre-trained on Imagenet
+            trainable_backbone_layers: number of trainable resnet layers starting from final block
+        """
+        super().__init__()
+        self.learning_rate = learning_rate
+        self.num_classes = num_classes
+        self.backbone = backbone
```

Comment:
why not `save_hyperparameters()`?

Reply:
Since I shifted the metrics, the args have changed.

I'm not sure how `save_hyperparameters()` works.

Reply:
@oke-aditya Hope this example helps:
```python
import pytorch_lightning as pl

class Model(pl.LightningModule):
    def __init__(self, a, b, c):
        super().__init__()
        self.save_hyperparameters()
    def print(self):
        print(self.hparams, type(self.hparams))
        print(self.hparams.a, self.hparams.b, self.hparams.c)

model = Model(1, 2, 3)
model.print()
```
Output:
```
"a": 1
"b": 2
"c": 3 <class 'pytorch_lightning.utilities.parsing.AttributeDict'>
1 2 3
```